windows/src/desktop/kmshell/install/Keyman.Configuration.System.TIPMaintenance.pas

```diff
@@ -199,8 +202,58 @@ class function TTIPMaintenance.DoRegister(const KeyboardID,
 end;
 
 class function TTIPMaintenance.GetUserDefaultLanguage: string;
+var
+  r: TRegistry;
+  tags: TStringList;
+  v: string;
+  keys: TStringList;
+  key: string;
 begin
-  Result := TLanguageCodeUtils.TranslateWindowsLanguagesToBCP47(HKLToLanguageID(GetDefaultHKL))
+  // Fallback result
+  Result := TLanguageCodeUtils.TranslateWindowsLanguagesToBCP47(HKLToLanguageID(GetDefaultHKL));
+
+  // For Win10, look in CPL/International/UserProfile
+  r := TRegistry.Create;
+  try
+    if not r.OpenKeyReadOnly(SRegKey_ControlPanelInternationalUserProfile) then
+      Exit;
+    if r.ValueExists(SRegValue_CPIUP_InputMethodOverride) then
+    begin
+      // Lookup the override input method BCP 47 tag
```

Comment:
What sets this?

Reply:
This is set by Windows.

Reply:
So Windows is using BCP-47 tags here?  No need for `TLanguageCodeUtils.TranslateWindowsLanguagesToBCP47()`?

Reply:
Yes. The new language profile model uses BCP 47 tags which is fantastic. It encapsulates the legacy profile model which doesn't use BCP 47 (and this encapsulation gives us beauties such as transient profiles), but the more we can work with the new BCP 47-based language profiles, the better.

Reply:
Might be nice to have an inline comment indicating this aspect of the Win API functions (when a function uses legacy vs BCP-47), making it clearer which functions do and don't need the util function.  But... okay.